packages/turf-clusters-kmeans/index.js

```diff
@@ -27,37 +23,36 @@ var featureCollection = helpers.featureCollection;
  * var clustered = turf.clustersKmeans(points, numberOfClusters);
  *
  * //addToMap
- * var addToMap = featureCollection(clustered.points);
+ * var addToMap = clustered;
  */
 module.exports = function (points, numberOfClusters) {
     // Input validation
     collectionOf(points, 'Point', 'Input must contain Points');
+
     // Default Params
     var count = points.features.length;
-    if (numberOfClusters > count) throw new Error('numberOfClusters can\'t be grated than the number of points');
     numberOfClusters = numberOfClusters || Math.round(Math.sqrt(count / 2));
 
+    // numberOfClusters can't be greater than the number of points
+    // fallbacks to count
+    if (numberOfClusters > count) numberOfClusters = count;
+
+    // Clone points to prevent any mutations
+    points = clone(points, true);
```

Comment:
Shouldn't we let the user decide if the input should change or (by default) not, like we did for the `transform` modules? I believe it's likely the user _wants_ the input to be added the clustering properties